app/src/AddJoke.tsx

```diff
@@ -1,27 +1,78 @@
-import React, { useState } from "react";
+import { useState, ChangeEvent } from "react";
+import { AppliedTheFavouriteJoke } from "./AppliedTheFavouriteJoke";
+import { IJokes } from "./interfaces";
 
-export default function AddJoke() {
-  const [joke, setAddJoke] = useState("");
-  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
+interface AddJokeProps {
+  status: string;
+}
+
+export default function AddJoke({ status = "empty" }: AddJokeProps) {
+  const [joke, setJoke] = useState<string>("");
+  const [pun, setPun] = useState<string>("");
+  const [jokeList, setJokeList] = useState<IJokes[]>([]);
```

Comment:
I see so the IJokes empty array  interface  gets passed down as a starting point for jokeList, makes a lot of sense and is very clear ✨ 🃏